Remotion/Web/Core/ContentSecurityPolicy/CspEnabledHtmlTextWriter.cs

```diff
@@ -91,37 +91,74 @@ public override void RenderBeginTag (HtmlTextWriterTag tagKey)
       base.RenderBeginTag(tagKey);
     }
 
-    protected override void AddAttribute (string name, string? value, HtmlTextWriterAttribute key, bool encode, bool isUrl)
+    public sealed override void AddAttribute (string name, string? value)
     {
       ArgumentUtility.CheckNotNull("name", name);
+      if (!TryAddAttributeWithoutEncoding(name, value, false))
+        base.AddAttribute(name, value);
+    }
 
-      if (name.StartsWith("on", StringComparison.OrdinalIgnoreCase) && !string.IsNullOrEmpty(value))
-      {
-        if (s_supportedEvents.TryGetValue(name, out var eventType))
-        {
-          if (_registeredEvents.Exists(e => eventType.Equals(e.Type)))
-          {
-            throw new ArgumentException($"Event handler '{name}' cannot be registered more than once.");
-          }
+    public sealed override void AddAttribute (string name, string? value, bool encode)
+    {
+      ArgumentUtility.CheckNotNull("name", name);
+      if (!TryAddAttributeWithoutEncoding(name, value, !encode))
+        base.AddAttribute(name, value, encode);
+    }
 
-          var trimmedValue = value.TrimStart();
-          const string javascriptPrefix = "javascript:";
-          if (trimmedValue.StartsWith(javascriptPrefix, StringComparison.OrdinalIgnoreCase))
-            value = trimmedValue.Substring(javascriptPrefix.Length).TrimStart();
+    protected sealed override void AddAttribute (string name, string? value, HtmlTextWriterAttribute key)
+    {
+      base.AddAttribute(name, value, key);
+    }
 
-          _registeredEvents.Add((Type: eventType, Value: value));
-        }
-        else
+    protected sealed override void AddAttribute (string name, string? value, HtmlTextWriterAttribute key, bool encode, bool isUrl)
+    {
+      ArgumentUtility.CheckNotNull("name", name);
+      if (!TryAddAttributeWithoutEncoding(name, value, !encode))
+        base.AddAttribute(name, value!, key, encode, isUrl);
+    }
+
+    public sealed override void AddAttribute (HtmlTextWriterAttribute key, string? value)
+    {
+      base.AddAttribute(key, value);
+    }
+
+    public sealed override void AddAttribute (HtmlTextWriterAttribute key, string? value, bool fEncode)
+    {
+      base.AddAttribute(key, value, fEncode);
+    }
+
+    private bool TryAddAttributeWithoutEncoding (string name, string? value, bool isAlreadyEncoded)
+    {
+      ArgumentUtility.CheckNotNull("name", name);
+
+      if (!name.StartsWith("on", StringComparison.OrdinalIgnoreCase) || string.IsNullOrEmpty(value))
+        return false;
+
```

Comment:
```suggestion
      if (!name.StartsWith("on", StringComparison.OrdinalIgnoreCase))
        return false;

      if (string.IsNullOrEmpty(value))
        return false;

```
I prefer to split up OR-d IFs with a simple return value (i.e. exit conditions). better test coverage and easier to read.